mne/io/edf/edf.py

```diff
@@ -511,16 +490,511 @@ def _get_edf_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
+        units = [fid.read(8).strip().decode() for ch in channels]
+        edf_info['units'] = list()
+        include = list()
+        for i, unit in enumerate(units):
+            if i in exclude:
+                continue
+            if unit == 'uV':
+                edf_info['units'].append(1e-6)
+            else:
+                edf_info['units'].append(1)
+            include.append(i)
+        ch_names = [ch_names[idx] for idx in include]
+
+        physical_min = np.array([float(fid.read(8).decode())
+                                 for ch in channels])[include]
+        physical_max = np.array([float(fid.read(8).decode())
+                                 for ch in channels])[include]
+        digital_min = np.array([float(fid.read(8).decode())
+                                for ch in channels])[include]
+        digital_max = np.array([float(fid.read(8).decode())
+                                for ch in channels])[include]
+        prefiltering = [fid.read(80).decode().strip(' \x00')
+                        for ch in channels][:-1]
+        highpass = np.ravel([re.findall('HP:\s+(\w+)', filt)
+                             for filt in prefiltering])
+        lowpass = np.ravel([re.findall('LP:\s+(\w+)', filt)
+                            for filt in prefiltering])
+
+        # number of samples per record
+        n_samps = np.array([int(fid.read(8).decode()) for ch
+                            in channels])
+
+        # Populate edf_info
+        edf_info.update(
+            ch_names=ch_names, data_offset=header_nbytes,
+            digital_max=digital_max, digital_min=digital_min, exclude=exclude,
+            highpass=highpass, include=include, lowpass=lowpass,
+            meas_date=calendar.timegm(date.utctimetuple()),
+            n_records=n_records, n_samps=n_samps, nchan=nchan,
+            subject_info=patient, physical_max=physical_max,
+            physical_min=physical_min, record_length=record_length,
+            subtype=subtype)
+
+        fid.read(32 * nchan).decode()  # reserved
+        assert fid.tell() == header_nbytes
+
+        fid.seek(0, 2)
+        n_bytes = fid.tell()
+        n_data_bytes = n_bytes - header_nbytes
+        total_samps = (n_data_bytes // 3 if subtype == '24BIT'
+                       else n_data_bytes // 2)
+        read_records = total_samps // np.sum(n_samps)
+        if n_records != read_records:
+            warn('Number of records from the header does not match the file '
+                 'size (perhaps the recording was not stopped before exiting).'
+                 ' Inferring from the file size.')
+            edf_info['n_records'] = n_records = read_records
+
+        if subtype in ('24BIT', 'bdf'):
+            edf_info['dtype_byte'] = 3  # 24-bit (3 byte) integers
+            edf_info['dtype_np'] = np.uint8
+        else:
+            edf_info['dtype_byte'] = 2  # 16-bit (2 byte) integers
+            edf_info['dtype_np'] = np.int16
```

Comment:
I believe it's little endian by default. `<i2` is what the previous code was doing.

```
>>> dt = np.dtype('<i2')
>>> print dt
int16
>>> dt = np.dtype('>i2')
>>> print dt
>i2
>>> dt = np.dtype(np.int16)
>>> print dt
int16
```